packages/components/date-picker/src/date-picker-com/panel-year-range.vue

```diff
@@ -0,0 +1,263 @@
+            @pick="handleRangePick"
+            @select="onSelect"
+          />
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+<script lang="ts" setup>
+import { computed, inject, ref, toRef, watch } from 'vue'
+import dayjs from 'dayjs'
+import { DArrowLeft, DArrowRight } from '@element-plus/icons-vue'
+import ElIcon from '@element-plus/components/icon'
+import { useLocale, useNamespace } from '@element-plus/hooks'
+import {
+  panelYearRangeEmits,
+  panelYearRangeProps,
+} from '../props/panel-year-range'
+import { useShortcut } from '../composables/use-shortcut'
+import { useYearRangeHeader } from '../composables/use-year-range-header'
+import { isValidRange } from '../utils'
+import { ROOT_PICKER_INJECTION_KEY } from '../constants'
+import YearTable from './basic-year-table.vue'
+
+import type { Dayjs } from 'dayjs'
+import type { RangeState } from '../props/shared'
+
+defineOptions({
+  name: 'DatePickerYearRange',
+})
+
+const props = defineProps(panelYearRangeProps)
+const emit = defineEmits(panelYearRangeEmits)
+
+const { lang } = useLocale()
+const leftDate = ref(dayjs().locale(lang.value))
+const rightDate = ref(leftDate.value.add(10, 'year'))
+const { pickerNs: ppNs } = inject(ROOT_PICKER_INJECTION_KEY)!
+const drpNs = useNamespace('date-range-picker')
+
+const hasShortcuts = computed(() => !!shortcuts.length)
+
+const handleShortcutClick = useShortcut(lang)
+
+const {
+  leftPrevYear,
+  rightNextYear,
+  leftNextYear,
+  rightPrevYear,
+  leftLabel,
+  rightLabel,
+  leftYear,
+  rightYear,
+} = useYearRangeHeader({
+  unlinkPanels: toRef(props, 'unlinkPanels'),
+  leftDate,
+  rightDate,
+})
+
+const enableYearArrow = computed(() => {
+  return props.unlinkPanels && rightYear.value > leftYear.value + 1
+})
+
+const minDate = ref<Dayjs>()
+const maxDate = ref<Dayjs>()
+
+const rangeState = ref<RangeState>({
+  endDate: null,
+  selecting: false,
+})
+
+const handleChangeRange = (val: RangeState) => {
+  rangeState.value = val
+}
+
+type RangePickValue = {
+  minDate: Dayjs
+  maxDate: Dayjs
+}
+const handleRangePick = (val: RangePickValue, close = true) => {
+  const minDate_ = val.minDate
+  const maxDate_ = val.maxDate
+  if (maxDate.value === maxDate_ && minDate.value === minDate_) {
+    return
+  }
+  emit('calendar-change', [minDate_.toDate(), maxDate_ && maxDate_.toDate()])
+  maxDate.value = maxDate_
+  minDate.value = minDate_
+
+  if (!close) return
+  handleConfirm()
+}
+
+const handleConfirm = (visible = false) => {
+  if (isValidRange([minDate.value, maxDate.value])) {
+    emit('pick', [minDate.value, maxDate.value], visible)
+  }
+}
+
+const onSelect = (selecting: boolean) => {
+  rangeState.value.selecting = selecting
+  if (!selecting) {
+    rangeState.value.endDate = null
+  }
+}
+
+const formatToString = (days: Dayjs[]) => {
+  return days.map((day) => day.format(format))
+}
+const getDefaultValue = () => {
+  let start: Dayjs
+  if (Array.isArray(defaultValue.value)) {
```

Comment:
Use encapsulated methods.
```suggestion
  if (isArray(defaultValue.value)) {
```

Reply:
Have import isArray from @element-plus/utils